### PR TITLE
Upgrade to terra-functional-testing and Node 12

### LINKED ISSUE #138: Node 12 Upgrade
# Story

## Description

As a maintainer I would like to upgrade this component to Node 12 (lts/erbium)

## Motivation

So that it can have the latest features and security updates and remain tech current with Node.

## Definition of Done

We know we're done when:

- [ ] The .nvmrc file is using lts/fermium
- [ ] All tests are passing
- [ ] A PR is merged to the main branch

## Additional Details
The recently released [version 7](https://webdriver.io/blog/2021/02/09/webdriverio-v7-released) of WebDriverIO dropped support for Node 10 (Dubnium) and recommends to be on Node 12 (Erbium) or higher. The [support|https://github.com/nodejs/release#release-schedule]] for Node 10 will end on 04/30/2021.

### LINKED ISSUE #208: Upgrade terra-application to terra-functional-testing
# Story

## Description

As a developer of terra I want to upgrade packages to terra-functional-testing

## Motivation

So that we can get the latest and greatest features of WDIO 6.

## Definition of Done

We'll know we're done when

All packages in repo have wdio testing enabled
All tests pass
PR has been merged into main

### PR DESCRIPTION
**Pull request by benbcai**
### Summary
Upgrade to terra-functional-testing and Node 12.

Closes #208 
Closes #138 

### Deployment Link

https://terra-applic-.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "version": "1.47.1",
   "description": "A framework to support application development with Terra components",
   "engines": {
-    "node": ">=8.9.2 <11"
+    "node": ">=10"
```
Comment (mjhenkes) [inline package.json]: I think you have to specify less than 16 to build on heroku
```suggestion
    "node": ">=10 <16"
```

**Review by mjhenkes — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "version": "1.47.1",
   "description": "A framework to support application development with Terra components",
   "engines": {
-    "node": ">=8.9.2 <11"
+    "node": ">=10 <16"
```
Comment (mjhenkes) [inline package.json]: lets try less than 15, npm 7 doesn't much care for our peer dependency issues.
```suggestion
    "node": ">=10 <15"
```

**Review by dkasper-was-taken — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "version": "1.47.1",
   "description": "A framework to support application development with Terra components",
   "engines": {
-    "node": ">=8.9.2 <11"
+    "node": ">=10 <15"
```
Comment (pranav300) [inline package.json:7]: Shouldn't this be `<13`? Apart from this everything looks good.

Comment (benbcai) [reply package.json:7]: terra-application can also support Node 14 so we want to include 14. Just not 16 yet.

**Review by pranav300 — APPROVED** [review]

**Pull request merged**